updater/lib/dependabot/updater/operations/create_security_update_pull_request.rb

```diff
@@ -59,17 +59,22 @@ def initialize(service:, job:, dependency_snapshot:, error_handler:)
         def perform
           Dependabot.logger.info("Starting security update job for #{job.source.repo}")
 
+          target_dependencies = dependency_snapshot.job_dependencies
+
+          # Raise an error if the package manager version is unsupported
+          dependency_snapshot.package_manager&.raise_if_unsupported!
```

Comment:
When we have `unsupported error` for the package manager, we don't want to go over each dependency, instead we want to raise unsupported error for the whole process and handle error properly.

Reply:
It would be simpler to not modify these operations, and have `DependencySnapshot::dependency_file_parser` raise an error right when we calculate the "package_manager".

Reply:
I think the problem is we are not catching errors on top of the operations. So when I do that I also need to error handling. I will check the feasibility of that.

Reply:
@jakecoffman , 

is it ok to do that after cleaning-up the code. I think this will also need some tests and also after adding it to the `DependencySnapshot::dependency_file_parser` I need check error handling mechanism for this code the place where is called and handle error properly. Otherwise it is turning to unknown error.

Reply:
Just for posterity I raised a PR https://github.com/dependabot/dependabot-core/pull/10794 which is what I was suggesting.